pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -45,7 +45,15 @@
 warning_cache = WarningCache()
 
 
-class ModelCheckpoint(Callback):
+class BaseModelCheckpoint(Callback):
```

Comment:
Just so that someone commented about it: 
We have gradually moved away from naming base classes with the "Base" as pre- or postfix. Here of course, we are in the situation where we want to create a new base class and can't rename the existing one. Adding Base is probably the best here. Have other names be proposed? I can't think of a better one, all alternatives sound silly or are inconsistent: CheckpointCallback, Checkpoint, AbstractModelCheckpoint.

Reply:
I don't think other names have been proposed, but I agree with you that most of the alternatives sound a bit silly. The only one that sounds at least a bit reasonable is `AbstractModelCheckpoint` as it shows, that if you use just this without subclassing, you're not actually doing anything at all.

Reply:
I would only call it Abstract if it really is an abstract class (i.e. if an instance of this class cannot be created without subclassing. Otherwise i'd stick to base

Reply:
Actually, I like Checkpoint as the best option out of the selection we have. 
Checkpointer would be an outlier among all the other names (e.g. ModelSummary vs ModelSummarizer).

Reply:
Yeah, I like Checkpoint too. Applied